vyper/compiler/output.py

```diff
@@ -254,7 +254,8 @@ def build_method_identifiers_output(compiler_data: CompilerData) -> dict:
 
 def build_abi_output(compiler_data: CompilerData) -> list:
     module_t = compiler_data.annotated_vyper_module._metadata["type"]
-    _ = compiler_data.ir_runtime  # ensure _ir_info is generated
+    if compiler_data.contract_path.suffix != ".vyi":
```

Comment:
maybe we should tag it on the ModuleT or Module node object somehow

Reply:
I've added it to slots in Module- not sure that is the best.
The argument `is_interface` is added to `parse_to_ast_with_setting `instead of `parse_to_ast` because the `_generate_ast` calls the `parse_to_ast_with_settings` funciton directly.

Reply:
yes, it looks great